frontend/lib/src/components/elements/Expander/Expander.tsx

```diff
@@ -22,13 +22,85 @@ import {
   SharedStylePropsArg,
 } from "baseui/accordion"
 import { useTheme } from "@emotion/react"
-import { ExpandMore, ExpandLess } from "@emotion-icons/material-outlined"
-import { Block as BlockProto } from "@streamlit/lib/src/proto"
+import {
+  ExpandMore,
+  ExpandLess,
+  Check,
+  ErrorOutline,
+} from "@emotion-icons/material-outlined"
 
-import Icon from "@streamlit/lib/src/components/shared/Icon"
+import { Block as BlockProto } from "@streamlit/lib/src/proto"
+import {
+  StyledSpinnerIcon,
+  StyledIcon,
+} from "@streamlit/lib/src/components/shared/Icon"
 import StreamlitMarkdown from "@streamlit/lib/src/components/shared/StreamlitMarkdown"
+import { notNullOrUndefined } from "@streamlit/lib/src/util/utils"
+import { LibContext } from "@streamlit/lib/src/components/core/LibContext"
+import { IconSize, isPresetTheme } from "@streamlit/lib/src/theme"
+
+import {
+  StyledExpandableContainer,
+  StyledIconContainer,
+} from "./styled-components"
+
+export interface ExpanderIconProps {
+  icon: string
+}
 
-import { StyledExpandableContainer } from "./styled-components"
+/**
+ * Renders an icon for the expander.
+ *
+ * If the icon is "spinner", it will render a spinner icon.
+ * If the icon is "check", it will render a check icon.
+ * If the icon is "error", it will render an error icon.
+ * Otherwise, it will render nothing.
+ *
+ * @param {string} icon - The icon to render.
+ * @returns {ReactElement}
+ */
+export const ExpanderIcon = (props: ExpanderIconProps): ReactElement => {
```

Comment:
should this be a separate component in its own separate file?

Reply:
I actually initially created a more generic version of this in the Icon component, so it can also be reused in other components (everything where we allow icons in streamlit). And it's probably the best long-term place for this. But I think for now, it's just a very specific subpart of the Expander that doesn't have a lot of logic on its own, so I would probably leave it here for now.